### PR TITLE
Fixes based on feedback for the original #59929

### PR DESCRIPTION
**Pull request by vitek-karas**
Adds apphost_version to the URL (most important fix)
Adds version info to the error message for 6.0

Cleans up some stuff in tests

### CODE REVIEW

File: src/native/corehost/test/mockhostfxr/mockhostfxr.cpp [hunk 1]
```diff
@@ -20,7 +20,7 @@ SHARED_API int HOSTFXR_CALLTYPE hostfxr_main_startupinfo(const int argc, const p
 
     const pal::string_t dotnet_folder = get_directory(dotnet_root);
```
Comment (elinor-fung) [inline src/native/corehost/test/mockhostfxr/mockhostfxr.cpp:21]: `dotnet_root` should already be the folder. I guess that would explain why the tests have been working without `== 0 ` in the condition 😬

File: src/native/corehost/test/mockhostfxr/mockhostfxr.cpp [hunk 2]
```diff
@@ -40,7 +40,7 @@ SHARED_API int HOSTFXR_CALLTYPE hostfxr_main_bundle_startupinfo(const int argc,
...
```
Comment (elinor-fung) [inline src/native/corehost/test/mockhostfxr/mockhostfxr.cpp:41]:
```suggestion
    const pal::string_t dotnet_folder = get_filename(dotnet_root);
```

**Comment by vitek-karas** [comment]
CI failure is #59908

**Review by elinor-fung — APPROVED** [review]

**Review by agocke — APPROVED** [review]
LGTM

**Pull request merged**